### PR TITLE
MH-13581, Streaming Module Cleanup

### PR DESCRIPTION
**Pull request by lkiesow**
The adaptive streaming module of Opencast contains a lot of unnecessary,
outdated or somewhat risky code. This patch strives to improve the
module by:

- letting the module activation fail on invalid configurations
- removing duplicated code and checks
- removing unnecessary array/list conversions

---

@mliradelc, since you have an actual streaming server set-up and I don't, would youn mind testing this under real conditions?

### CODE REVIEW

**Comment by KatrinIhler** [comment]
@mliradelc What's the status here? Do you want to review this?

**Comment by uzk-rrzk** [comment]
Today I back ported the changes to version 7.x. The data were successfully created the first time, but the retract operation failed and as a result also a republish of the workflow that was started in the video editor.
These new lines seem to be the problem. 

```
  if (distributionDirectory != null && (streamingUri != null || adaptiveStreamingUri != null)) {
      logger.warn("Invalid configuration");
      return Collections.emptyList();
    }

```

**Comment by gregorydlogan** [comment]
@lkiesow Are you going to have a chance to move this forward any time soon?

**Comment by lkiesow** [comment]
Probably, yes.

File: modules/distribution-service-adaptive-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaAdaptiveStreamingDistributionService.java [hunk 1]
```diff
@@ -452,102 +443,82 @@ public Job distribute(String channelId, MediaPackage mediapackage, Set<String> e
...
-    final MediaPackageElement element = mediapackage.getElementById(elementId);
+  private synchronized List<MediaPackageElement> distributeElement(final String channelId,
+          final MediaPackage mediaPackage, final MediaPackageElement element) throws DistributionException {
 
-    // Make sure the element exists
-    if (element == null)
-      throw new IllegalStateException(
-              "No element " + elementId + " found in mediapackage" + mediapackage.getIdentifier());
+    if (!isRTMPSupported && supportedAdaptiveFormats.isEmpty()) {
+      logger.warn("Skipping distribution of element \"{}\" because no streaming format was specified", element);
```
Comment (gregorydlogan) [inline modules/distribution-service-adaptive-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaAdaptiveStreamingDistributionService.java:492]: I feel like this should be a `DistributionException`...

**Pull request merged**